File: Stack & Queue/000 Stack/Valid Parentheses/Solution.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack();
        st.add(s.charAt(0));
        for(int i=1;i<s.length();i++) {
            char temp= s.charAt(i);
            if(temp== '(' || temp== '{' || temp== '[')
                st.add(temp);
            
            if(st.isEmpty())
                return false;
            
            switch(temp) {
                case ')' : if(st.pop()!='(')
                                return false;
                           break;
                case '}' : if(st.pop()!='{')
                                return false;
                           break;
                case ']' : if(st.pop()!='[')
                                return false;
            }
        }
        
        return st.isEmpty();
    }
}

File: Stack & Queue/000 Stack/Valid Parentheses/Solution.kt
class Solution {
    fun isValid(s: String): Boolean {
        val st = Stack<Char>()
        
        for(c in s)
            when(c) {
                '(' -> st!!.add(')')
                '{' -> st!!.add('}')
                '[' -> st!!.add(']')
                else ->  if(st.isEmpty() || st!!.pop()!=c)
                                return false
            }
        
        return st.isEmpty()
    }
}
